### PR TITLE
feat: Test cdk synth

### PR DESCRIPTION
**Pull request by Jimlinz**
Ensure CDK can be synthesized.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,16 +1,16 @@
 version: 2
 updates:
-  - package-ecosystem: "github-actions"
-    directory: "/"
+  - package-ecosystem: 'github-actions'
+    directory: '/'
```
Comment (l0b0) [inline .github/dependabot.yml:4]: There's a bunch of formatting changes like this in the PR. They should probably be in a separate PR with the relevant configuration.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,36 +15,70 @@ jobs:
...
       - name: Install Poetry
         run: python -m pip install poetry
+
       - name: Install Python dependencies
         run: python -m poetry install
+
+      - name: Use Node.js for CDK deployment
+        uses: actions/setup-node@v3.5.1
+        with:
+          node-version: 14
```
Comment (l0b0) [inline .github/workflows/ci.yml:50]: setup-node supports [getting the version from a file](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file), which means tools with support for `.nvmrc` could use the same version seamlessly. Also, 18.12.1 seems to be the latest stable Node.js version.

File: .github/dependabot.yml (hunk 1 above)
Comment (Jimlinz) [reply .github/dependabot.yml:4]: Have moved to https://github.com/linz/oidc-provider/pull/100

**Review by l0b0 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -12,41 +12,73 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          - ubuntu-18.04
           - ubuntu-20.04
```
Comment (l0b0) [inline .github/workflows/ci.yml:15]: Do we need to run this on more than one Ubuntu release?

Comment (Jimlinz) [reply .github/workflows/ci.yml]: This was probably carried over from the template. Testing on multiple releases could be beneficial since it could fail on older releases still within LTS. Having said that, the overhead of maintaining compatibility with multiple OS's and versions is rather high for a stack that we are likely to only deploy once per project. Have removed and pin to latest ubuntu lts release

**Review by l0b0 — APPROVED** [review]
